module_utils/configuration.py

```diff
@@ -431,7 +431,9 @@ def raise_for_failure(resp):
                                            path_params=path_params, query_params=query_params)
         raise_for_failure(response)
 
-        if http_method != HTTPMethod.GET and response[ResponseParams.STATUS_CODE] != NO_CONTENT_STATUS:
+        config_not_changed = http_method == HTTPMethod.GET or (
+                http_method != HTTPMethod.DELETE and response[ResponseParams.STATUS_CODE] == NO_CONTENT_STATUS)
+        if not config_not_changed:
```

Comment:
Please replace double negation with positive condition for better readability.

Reply:
Updated.